docs/DeveloperGuide.md

```diff
@@ -1,25 +1,88 @@
 # Developer Guide
 
-## Design & implementation
+## Design 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Project overview
```

Comment:
Is it better to add more pictures or represent ideas in point form so that your overview could be less wordy?